### PR TITLE
v1.13 Backports 2024-04-02

### PR DESCRIPTION
**Pull request by joamaki**
* [x] #29300 (@learnitall)
 * [x] #31488 (@bimmlerd) (:warning:  bunch of conflicts solved, please check)
 * [x] #31478 (@julianwiedmann)
 * [x] #31635 (@pchaigno)
 * [x] #31622 (@gandro)
 * [x] #31627 (@pchaigno)
 * [x] #31641 (@tklauser) (:warning: please check. digests were missing from other images, should we add? renovate commit a no-op?)
 * [x] #31652 (@qmonnet)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 29300 31488 31478 31635 31622 31627 31641 31652
```

### CODE REVIEW

**Comment by joamaki** [comment]
/test-backport-1.13

File: pkg/endpointmanager/gc_test.go [hunk 1]
```diff
@@ -39,6 +39,12 @@ func (s *EndpointManagerSuite) TestmarkAndSweep(c *C) {
 	for _, id := range allEndpointIDs {
 		ep := endpoint.NewEndpointWithState(s, s, testipcache.NewMockIPCache(), &endpoint.FakeEndpointProxy{}, testidentity.NewMockIdentityAllocator(nil), id, endpoint.StateReady)
 		mgr.expose(ep)
+
+		/*=======
```
Comment (bimmlerd) [inline pkg/endpointmanager/gc_test.go]: conflict left in tree-

mostly relevant to check `expose` err

**Review by bimmlerd — CHANGES_REQUESTED** [review]
conflict left in tree

File: pkg/endpointmanager/gc_test.go (hunk 1 above)
Comment (joamaki) [reply pkg/endpointmanager/gc_test.go]: Fixed.

**Review by bimmlerd — APPROVED** [review]
thanks!

**Comment by joamaki** [comment]
/test-backport-1.13

**Review by pchaigno — CHANGES_REQUESTED** [review]
The backport for https://github.com/cilium/cilium/pull/31627 is empty because that file moved. The code to edit in v1.13 is at https://github.com/cilium/cilium/blob/v1.13/.github/workflows/conformance-ipsec-e2e.yaml#L338.

**Review by qmonnet — APPROVED** [review]
Thanks!

**Review by tklauser — APPROVED** [review]
My change looks good, thanks.

> (:warning: please check. digests were missing from other images, should we add? renovate commit a no-op?)

We should probably add these digests but let's do this in separate backports. The renovate commit shouldn't be a no-op because `.github/renovate.json5` on `main` also covers all the stable branches and should thus also update that image based on the renovate comment added in commit "[install/kubernetes: use renovate to update quay.io/cilium/startup-script](https://github.com/cilium/cilium/pull/31722/commits/6e6939a2608582f437cfebf2390f5657746f64ec)"

**Review by learnitall — APPROVED** [review]
Thank you!

File: .github/workflows/conformance-ipsec-e2e.yaml [hunk 2]
```diff
@@ -335,7 +335,7 @@ jobs:
               if [[ $keys_in_use == $exp_nb_keys ]]; then
                 break
               fi
-              echo "Waiting until key rotation starts (seeing $keys_in_use keys)"
+              echo "Waiting until key rotation completes (seeing $keys_in_use keys, expected $exp_nb_keys)"
```
Comment (pchaigno) [inline .github/workflows/conformance-ipsec-e2e.yaml]: It looks like `starts` was changed into `completes` by mistake here.

Comment (joamaki) [reply .github/workflows/conformance-ipsec-e2e.yaml]: Oopsie, fixed.

**Review by pchaigno — APPROVED** [review]
All good. Thank you!

**Review by julianwiedmann — APPROVED** [review]

**Comment by joamaki** [comment]
/test-backport-1.13

**Review by tklauser — APPROVED** [review]
LGTM for #31622 as well, thanks.

**Comment by joamaki** [comment]
/test-backport-1.13

File: .github/workflows/conformance-e2e.yaml [hunk 3]
```diff
@@ -194,6 +194,19 @@ jobs:
...
+            kube-proxy: 'none'
+            kpr: 'strict'
+            devices: '{eth0,eth1}'
+            secondary-network: 'true'
+            tunnel: 'vxlan'
+            lb-mode: 'snat'
+            encryption: 'wireguard'
+            encryption-node: 'false'
+            host-fw: 'true'
+
```
Comment (julianwiedmann) [inline .github/workflows/conformance-e2e.yaml]: @qmonnet this seems to fail reliably in CI for
> north-south-loadbalancing/outside-to-nodeport/curl-0: cilium-test/host-netns-non-cilium-j8jpw (172.18.0.4) -> cilium-test/echo-same-node (echo-same-node.cilium-test:8080)

Let's maybe just drop for now (wireguard on v1.13 isn't very relevant) ?

Comment (qmonnet) [reply .github/workflows/conformance-e2e.yaml]: Yes that's fine by me, let's drop it from this branch. Thanks!

Comment (joamaki) [reply .github/workflows/conformance-e2e.yaml]: Dropped it

**Comment by joamaki** [comment]
/test-backport-1.13

**Pull request merged**